ksqldb-engine/src/main/java/io/confluent/ksql/query/MaterializationProviderBuilderFactory.java

```diff
@@ -101,6 +106,19 @@ public MaterializationProviderBuilder materializationProviderBuilder(
         Optional.empty()
     ).serializer();
 
+    if (!ksqlConfig.getBoolean(KsqlConfig.KSQL_SHARED_RUNTIME_ENABLED)) {
```

Comment:
what does this have to do with whether or not the shared runtime is enabled?

Reply:
Add same logic to the shared runtime code path.